pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -221,19 +233,23 @@ func (r *Reconciler) fetchNamespaceResources(ctx context.Context, selector fleet
 }
 
 // fetchAllResourcesInOneNamespace retrieve all the objects inside a single namespace which includes the namespace itself.
-func (r *Reconciler) fetchAllResourcesInOneNamespace(_ context.Context, namespaceName string, placeName string) ([]runtime.Object, error) {
+func (r *Reconciler) fetchAllResourcesInOneNamespace(namespaceName string, placeName string) ([]runtime.Object, error) {
 	var resources []runtime.Object
 
 	if !utils.ShouldPropagateNamespace(namespaceName, r.SkippedNamespaces) {
-		return nil, fmt.Errorf("namespace %s is not allowed to propagate", namespaceName)
+		err := fmt.Errorf("invalid clusterRresourcePlacement %s: namespace %s is not allowed to propagate", placeName, namespaceName)
+		return nil, controller.NewClientError(err)
 	}
 
 	klog.V(2).InfoS("start to fetch all the resources inside a namespace", "namespace", namespaceName)
 	// select the namespace object itself
 	obj, err := r.InformerManager.Lister(utils.NamespaceGVR).Get(namespaceName)
 	if err != nil {
 		klog.ErrorS(err, "cannot get the namespace", "namespace", namespaceName)
-		return nil, client.IgnoreNotFound(err)
+		if errors.IsNotFound(err) {
```

Comment:
I do think a better way is to let the New function handle nil in a special manner.